src/libs/actions/MapboxTokens.js

```diff
@@ -0,0 +1,89 @@
+import _ from 'underscore';
+import moment from 'moment';
+import Onyx from 'react-native-onyx';
+import {AppState} from 'react-native';
+import ONYXKEYS from '../../ONYXKEYS';
+import * as API from '../API';
+import CONST from '../../CONST';
+
+let connectionID;
+let currentToken;
+let refreshTimeoutID;
+const refreshInterval = 1000 * 60 * 25;
+
+const refreshToken = () => {
+    console.debug('[MapboxTokens] refreshing token every 25 minutes', refreshInterval);
+
+    // Cancel any previous timeouts so that there is only one request to get a token at a time.
+    clearTimeout(refreshTimeoutID);
+
+    // Refresh the token every 25 minutes
+    refreshTimeoutID = setTimeout(() => {
+        console.debug('[MapboxTokens] Fetching a new token after waiting 25 minutes');
+        API.read('GetMapboxAccessToken');
+    }, refreshInterval);
+};
+
+const hasTokenExpired = () => {
+    const now = moment();
+    const expiration = moment(currentToken.expiration);
+    const minutesUntilTokenExpires = expiration.diff(now, 'minutes');
+    return minutesUntilTokenExpires < 0;
```

Comment:
Why compare based on minutes? Maybe we can do `return !now.isBefore(expiration)`?